### DISCUSSION TITLE
Wake up endpoint suddenly returning 406 - Not Acceptable. Has anybody else run into this?

### DISCUSSION THREAD

**Discussion by Rockster160** [timdorr/tesla-api#595] [Q&A]
Hey all! First off- thank you so much for reverse engineering the API and maintaining the documentation. It's amazing!

I set up a cool little system that allowed me to monitor my car. I had it pinging the API roughly every hour to get updates. This whole thing has been working for... A month? Everything has been great. However, suddenly the other day I started getting back a 406 when I try to hit the wake up endpoint. 

I believe I'm following all of the recommended times and back offs and retry criteria, but I'm wondering if I got blocked because of the frequent requests? All of the auth / refresh endpoints are working as expected. I can even hit /vehicles to get that data, but trying to hit any endpoints specific to my car (Such as `latest_vehicle_data`, or any commands) results in an INSTANT 408 Timeout. Hitting the `wake_up` (which typically is what follows a timeout) endpoint results in the 406 - Not Acceptable.

The regular Tesla app is still working without issue. I've also tried regenerating my subscription code- all successful including auth until hitting an endpoint specific to my car. 

Any ideas for how to fix this, and more importantly- how to prevent it from happening in the future?

**Comment by rawmean** [comment]
Try this and see if you still get the 406 error.
```
curl --location --request POST 'https://owner-api.teslamotors.com/api/1/vehicles/your_vehicleID/wake_up' \
--header 'Authorization: Bearer your_token'
```

**Reply by Rockster160** [reply]
I tried that without the `--location` previously and it didn't work, but this time it worked... I don't see why a redirect would cause this, so I'm wondering if it's a one-off that it actually worked?

Now that it's awake, all commands are working, so I'll need to wait for it to go back to sleep to do some more testing. 
Things seem to work while the vehicle is awake, oddly enough. I wish the error messages had a little more explanation, though I understand it's Tesla's internal API, so they probably have internal docs that properly describe the behavior.

**Reply by rawmean** [reply]
I just tried it without the `--location` and while the car was `offline` and it still works.
Interestingly, if I use teslajson library (in python), I get the 406 error regardless of the online or sleep status. I'm trying to figure out what the difference is by the process of elimination.

Also, this always works:
```
   import requests
     url = "https://owner-api.teslamotors.com/api/1/vehicles/my_vid/wake_up"  
   payload = ""
    headers = {   'Authorization': 'Bearer my_token' }
     response = requests.request("POST", url, headers=headers, data=payload)  
   print(response.text)
```

**Reply by Rockster160** [reply]
I think I've got it figured it out. At least on my side, it looks like RestClient (Rails) was automatically changing the content-type to `"application/x-www-form-urlencoded"` because of the empty payload, despite it being manually set to json. They (Tesla) must have recently swapped things out so that the content type is required to be `"application/json"` or it throws the 406- which actually makes sense. 

So anybody that has the problem going forward, double check that your tool is not secretly changing your content type. You can use something like RequestBin to analyze your headers and make sure things are going through the way you expect.

I pushed that up and things seem to be working again. Will report back if I run into any more issues. Thanks for the help, @rawmean!

**Reply by themonomers** [reply]
@Rockster160 I can confirm the same error and fix by setting content type to "application/json" works in Google Apps Script.

**Reply by dvermagithub** [reply]
Specific commands, not all, such as vent and close windows and changing charge limit started exhibiting a 406 error a few days ago.

Adding `'Content-type': 'application/json' ` to my python request command header did not initially work for me. Started getting an http 400 error response.

Solved that by fixing a few things in my code...first the data being passed to the request has to be formatted as proper json which means using **"** instead of **'** and the data has to be dumped into json format in python. kind of annoying but at least it is all working now.
Thanks for this thread.

**Reply by rawmean** [reply]
I can confirm that setting the payload as proper json is important. For me, adding the `'Content-type': 'application/json'` doesn't make a difference. Maybe the `request` module adds that automatically.
Example in python:

```
data={'percent': 85}
resp = requests.request('POST', full_url, headers=headers, data=json.dumps(data)) 
```

**Reply by heumat** [reply]
Have a look at #594, I just added a `"Accept": "application/json"` into the request header dict and it works now.

**Reply by dvermagithub** [reply]
@rawmean the code `data=json.dumps(data)` is essentially turning the payload into proper json format, which is what I had to add to my code in addition to `'Content-type': 'application/json'`

**Comment by heumat** [comment]
As written as a short [comment](https://github.com/timdorr/tesla-api/discussions/595#discussioncomment-3137995) on the earlier thread above, the following solved the `406` for me:

```
response = await session.get(
    url,
    headers={"authorization": f"Bearer {token}", "Accept": "application/json"},
    timeout=6,
)
```

credits to the guys in #594 !
